src/main.rs

```diff
@@ -176,33 +184,45 @@ fn run_shell(vm: &mut VirtualMachine) -> PyResult {
         println!("No previous history.");
     }
 
-    let ps1 = &objstr::get_value(&vm.sys_module.get_attr("ps1").unwrap());
-    let ps2 = &objstr::get_value(&vm.sys_module.get_attr("ps2").unwrap());
-    let mut prompt = ps1;
+    let mut prompt = get_prompt(vm, "ps1");
+
+    let mut continuing = false;
 
     loop {
-        match repl.readline(prompt) {
+        if !continuing {
+            prompt = get_prompt(vm, "ps1");
+        }
+        match repl.readline(&prompt) {
             Ok(line) => {
                 debug!("You entered {:?}", line);
                 input.push_str(&line);
-                input.push_str("\n");
+                input.push('\n');
                 repl.add_history_entry(line.trim_end());
 
+                if continuing {
+                    if line.is_empty() {
+                        continuing = false;
+                    } else {
+                        continue;
+                    }
+                }
+
                 match shell_exec(vm, &input, vars.clone()) {
                     Err(CompileError::Parse(ParseError::EOF(_))) => {
-                        prompt = ps2;
+                        prompt = get_prompt(vm, "ps2");
+                        continuing = true;
                         continue;
                     }
                     _ => {
-                        prompt = ps1;
                         input = String::new();
                     }
                 }
             }
             Err(ReadlineError::Interrupted) => {
                 // TODO: Raise a real KeyboardInterrupt exception
                 println!("^C");
-                break;
+                continuing = false;
```

Comment:
I think we should quit when hitting CTR+C, right? Now we continue.

Reply:
Oh, right. I should've discussed this before making the change, but usually in shells (iPython, bash/zsh/sh/sshshshshshs), ^C just cancels what you're currently typing, it doesn't completely end the shell like ^D does. I found it to be sort of annoying when testing something in the REPL, so I changed it here; if you disagree with it I can change it back though.

Reply:
Okay, in that case, nice job!